FrontEnd/src/UserAuthentication/FirestoreDB.js

```diff
@@ -66,30 +75,40 @@ const uploadPostImage = async (serviceCategory, imageUpload) => {
     const downloadURL = await getDownloadURL(imageRef);
     return downloadURL;
   } catch (error) {
-    throw new Error("Error uploading image");
+    throw new Error(`Error uploading image: ${error.message}`);
   }
 };
 
-const createPost = async (formData, imageUpload, userId) => {
+const createPost = async (formData, imageUpload, userId, userData) => {
   try {
-    const imageURL = await uploadPostImage(formData.serviceCategory, imageUpload);
-    const formDataWithImage = { ...formData, imageURL };
+    const imageURL = await uploadPostImage(
+      formData.serviceCategories[0],
+      imageUpload
+    );
+    const createdAt = Timestamp.now();
+    const vendorUsername = userData.UserName;
+    const formDataWithImage = {
+      ...formData,
+      imageURL,
+      createdAt,
+      vendorUsername,
+    };
 
     if (userId === null) {
```

Comment:
```suggestion
    if (userID === null) {
```

Reply:
I changed everything else but left the naming convention in fetchUserFeed because userID and userId are different